### PR TITLE
Query by facets as links

### PR DESCRIPTION
**Pull request by peteglondon**
# What
Allow rummager to index and query the following link types:

* facet_values
* facet_groups

As part of the work to address tech debt introduced by the [EU Exit business finder](https://www.gov.uk/find-eu-exit-guidance-business) and to [Improve the Business Finder tagging system](https://trello.com/c/zn5LBg3L/77-improve-the-business-finder-tagging-system).

We are introducing Facet Groups, Facets and FacetValues as content items.

Tagging to these things will then be done in `content-tagger` (similar to taxonomy) and implemented as links.

Trello: https://trello.com/c/RHl653OZ/245-rummager-query-on-facet-group-and-facet-value

### CODE REVIEW

**Comment by peteglondon** [comment]
[Mapping Rummager request/response in Finder Frontend](https://trello.com/c/Sd7omEFH/263-mapping-rummager-request-response-in-finder-frontend)

**Comment by peteglondon** [comment]
[Rummager query on facet group and facet value](https://trello.com/c/RHl653OZ/245-rummager-query-on-facet-group-and-facet-value)

File: lib/govuk_index/presenters/expanded_links_presenter.rb [hunk 1]
```diff
@@ -32,6 +32,10 @@ def taxons
       content_ids("taxons")
     end
 
+    def facet_groups
+      content_ids("facet_groups")
```
Comment (steventux) [inline lib/govuk_index/presenters/expanded_links_presenter.rb:36]: At the moment, the link from facet to facet group is in the `parent` links array https://github.com/alphagov/govuk-content-schemas/blob/master/formats/facet.jsonnet#L90 
https://github.com/alphagov/govuk-content-schemas/blob/master/examples/facet/publisher_v2/facet_links.json#L7
I'm not sure `facet_groups` is a valid link type.

**Review by steventux — CHANGES_REQUESTED** [review]
Looks good, one question around facet groups.

File: lib/govuk_index/presenters/expanded_links_presenter.rb (hunk 1 above)
Comment (steventux) [reply lib/govuk_index/presenters/expanded_links_presenter.rb:36]: Having discussed this we've agreed that content either needs `facet_groups` links or the facet group to be present in the `facet_values` links, the former is more explicit see: https://github.com/alphagov/govuk-content-schemas/commit/90be6c695ae3f5827790c68d10cee7b61f16c7ae

**Review by steventux — APPROVED** [review]
LGTM

**Pull request merged**